pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java

```diff
@@ -215,9 +215,14 @@ public static class ControllerPeriodicTasksConf {
         "controller.realtime.segment.deepStoreUploadRetryEnabled";
     public static final String DEEP_STORE_RETRY_UPLOAD_TIMEOUT_MS =
         "controller.realtime.segment.deepStoreUploadRetry.timeoutMs";
+    public static final String ENABLE_TMP_SEGMENT_ASYNC_DELETION =
+        "controller.realtime.split.commit.segment.tmp.cleanup.async.enable";
+    public static final String SPLIT_COMMIT_TMP_SEGMENT_LIFETIME_SECOND =
+        "controller.realtime.split.commit.segment.tmp.lifetime.second";
 
     public static final int MIN_INITIAL_DELAY_IN_SECONDS = 120;
     public static final int MAX_INITIAL_DELAY_IN_SECONDS = 300;
+    public static final int DEFAULT_SPLIT_COMMIT_TMP_SEGMENT_LIFETIME_SECOND = 600;
```

Comment:
I think we can keep this a bit higher (1 hour). In cases where controller is overwhelmed, deepstore latencies are high, etc. it is likely that the tmp segment may stick around for a good few minutes.

Reply:
The .tmp.<UUID> file is generated during segment commit phase. If there's any timeout during the remote file system connection + ZK update connection, then this .tmp.<UUID> won't be use anymore. The 10min comes from 5 min timeout in file system connection plus 5 min timeout in zk connection.

Reply:
It can be that the controller's thread-pool is full and the http-requests take a while to be processed because they get queued. The time that the tmp file lasts in that case could exceed 10 minutes or get close to it.

Reply:
make sense.